event-schema/event_schema.go

```diff
@@ -628,138 +667,121 @@ func (manager *EventSchemaManagerT) flushEventSchemas() {
-				_, err = txn.Exec(archiveVersionsSQL)
+		func() {
+			stmt, err := txn.Prepare(pq.CopyIn(SCHEMA_VERSIONS_TABLE,
+				"uuid", "event_model_id", "schema_hash", "schema", "metadata", "private_data",
+				"first_seen", "last_seen", "total_count"))
+			assertTxnError(err, txn)
+			defer func() { _ = stmt.Close() }()
+			for _, sv := range schemaVersionsInCache {
+				metadataJSON := getMetadataJSON(sv.reservoirSample, sv.SchemaHash)
+				privateDataJSON := getPrivateDataJSON(sv.SchemaHash)
+				sv.TotalCount = sv.reservoirSample.totalCount
+
+				_, err = stmt.Exec(sv.UUID, sv.EventModelID, sv.SchemaHash, string(sv.Schema),
+					string(metadataJSON), string(privateDataJSON), sv.FirstSeen, sv.LastSeen, sv.TotalCount)
 				assertTxnError(err, txn)
 			}
+			_, err = stmt.Exec()
+			assertTxnError(err, txn)
+		}()
 
-			func() {
-				stmt, err := txn.Prepare(pq.CopyIn(SCHEMA_VERSIONS_TABLE,
-					"uuid", "event_model_id", "schema_hash", "schema", "metadata", "private_data",
-					"first_seen", "last_seen", "total_count"))
-				assertTxnError(err, txn)
-				defer func() { _ = stmt.Close() }()
-				for _, sv := range schemaVersionsInCache {
-					metadataJSON := getMetadataJSON(sv.reservoirSample, sv.SchemaHash)
-					privateDataJSON := getPrivateDataJSON(sv.SchemaHash)
-					sv.TotalCount = sv.reservoirSample.totalCount
-
-					_, err = stmt.Exec(sv.UUID, sv.EventModelID, sv.SchemaHash, string(sv.Schema),
-						string(metadataJSON), string(privateDataJSON), sv.FirstSeen, sv.LastSeen, sv.TotalCount)
-					assertTxnError(err, txn)
-				}
-				_, err = stmt.Exec()
-				assertTxnError(err, txn)
-			}()
+		stats.NewTaggedStat(
+			"update_schema_version_count", stats.GaugeType, stats.Tags{"module": "event_schemas"},
+		).Gauge(len(versionIDs))
+		pkgLogger.Debugf("[EventSchemas][Flush] %d new schema versions", len(schemaVersionsInCache))
+	}
 
-			stats.NewTaggedStat(
-				"update_schema_version_count", stats.GaugeType, stats.Tags{"module": "event_schemas"},
-			).Gauge(len(versionIDs))
-			pkgLogger.Debugf("[EventSchemas][Flush] %d new schema versions", len(schemaVersionsInCache))
-		}
+	err = txn.Commit()
+	assertTxnError(err, txn)
 
-		err = txn.Commit()
-		assertTxnError(err, txn)
+	if err := flushDBHandle.Close(); err != nil {
+		pkgLogger.Warnf("Error flushing DB handle: %v", err)
+	}
 
-		if err := flushDBHandle.Close(); err != nil {
-			pkgLogger.Warnf("Error flushing DB handle: %v", err)
-		}
+	updatedEventModels = make(map[string]*EventModelT)
+	updatedSchemaVersions = make(map[string]*SchemaVersionT)
+	toDeleteEventModelIDs = []string{}
+	toDeleteSchemaVersionIDs = []string{}
 
-		updatedEventModels = make(map[string]*EventModelT)
-		updatedSchemaVersions = make(map[string]*SchemaVersionT)
-		toDeleteEventModelIDs = []string{}
-		toDeleteSchemaVersionIDs = []string{}
+	manager.schemaVersionLock.Unlock()
+	manager.eventModelLock.Unlock()
```

Comment:
Can we move these two on top as `defer`s right after:
```go
	if len(updatedEventModels) == 0 && len(schemaVersionsInCache) == 0 {
		manager.eventModelLock.Unlock()
		manager.schemaVersionLock.Unlock()
		return nil
	}
```

I'm asking this as to decrease the chances of messing up with the locks if someone where to add a return in between. It would also increase readability since we'll notice with a glance where we lock and where we unlock.